nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,487 @@
+
+    extractor: string {'connected_components', 'local_regions'}, optional
+        default is 'local_regions'
+        A string which chooses between the type of extractor.
+
+        If 'connected_components', regions are segmented using only a labels
+        assigned to each unique region/object.
+
+        If 'local_regions', regions are segmented using a seed points assigned by
+        its peak max value of that particular local regions.
+
+    smooth_fwhm: scalar, default smooth_fwhm=6. optional
+        smoothing parameter as a full width half maximum, in millimetres.
+        This scalar value is applied on all three directions (x,y,z).
+
+    Attributes
+    ----------
+    regions_: Niimg-like image/object
+        a list of seperated regions with each region lying on a 3D volume
+        concatenated into a 4D Nifti like object.
+
+    indentity_: a numpy array
+        an array of list of indices where each index value is assigned to
+        each seperate region of its corresponding family of brain maps.
+
+    signals_: a numpy array
+        a list of averaged timeseries signals of the subjects extracted from
+        each region.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask_img=None, target_affine=None,
+                 target_shape=None, standardize=False, low_pass=None,
+                 high_pass=None, t_r=None, memory=Memory(cachedir=None),
+                 min_size=20, thresholding_strategy='percentile',
+                 threshold='auto', extractor='local_regions',
+                 peak_local_smooth=6., verbose=0):
+        self.maps_img = maps_img
+        self.mask_img = mask_img
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.standardize = standardize
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+
+        # parameters for region extraction
+        self.min_size = min_size
+        self.thresholding_strategy = thresholding_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+        self.peak_local_smooth = peak_local_smooth
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """ Prepare or set up the data for the region extraction
+
+        """
+        maps_img = check_niimg_4d(self.maps_img)
+
+        # Asking NiftiMapsMasker to prepare the data for regions extraction
+        # only if mask_img is provided
+        if self.mask_img is not None:
+            nifti_maps_masker = NiftiMapsMasker(maps_img, mask_img=self.mask_img,
+                                                resampling_target="maps")
```

Comment:
Since you take the mask into account for region extraction, you don't need it here.